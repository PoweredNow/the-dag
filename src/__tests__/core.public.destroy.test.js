const addEdges = require('../core/public/addEdges');
const addNodes = require('../core/public/addNodes');
const destroy = require('../core/public/destroy');

const { getState } = require('../core/private/_state');
describe('core.public.destroy', () => {
  test('exports', () => {
    expect(destroy).toMatchSnapshot();
  });
  test('Destroys nodes and edges', () => {
    addNodes([{ nodeID: 1 }, { nodeID: 2 }]);
    addEdges([{ source: { nodeID: 1 }, target: { nodeID: 2 } }]);
    expect(getState()).toMatchSnapshot();

    destroy();

    const state = getState();
    expect(Object.keys(state.nodes).length).toBe(0);
    expect(Object.keys(state.edges).length).toBe(0);
    // expect(state).toMatchSnapshot();
  });
});
