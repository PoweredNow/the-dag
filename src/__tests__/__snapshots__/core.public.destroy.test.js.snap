// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core.public.destroy Destroys nodes and edges 1`] = `
Object {
  "edges": Object {
    "1_2": Object {
      "edgeData": Object {},
      "edgeID": "1_2",
      "sourceID": 1,
      "targetID": 2,
    },
  },
  "nodes": Object {
    "1": Object {
      "nodeData": undefined,
      "nodeID": 1,
      "possibleTargets": Array [
        2,
      ],
    },
    "2": Object {
      "nodeData": undefined,
      "nodeID": 2,
      "possibleSources": Array [
        1,
      ],
      "possibleTargets": Array [],
    },
  },
}
`;

exports[`core.public.destroy exports 1`] = `[Function]`;
